test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/MockCollectorIntegrationTests.cs

```diff
@@ -91,6 +98,197 @@ public async Task TestRecoveryAfterFailedExport()
         await host.StopAsync();
     }
 
+    [Theory]
+    [InlineData(true, ExportResult.Success, Grpc.Core.StatusCode.Unavailable)]
+    [InlineData(true, ExportResult.Success, Grpc.Core.StatusCode.Cancelled)]
+    [InlineData(true, ExportResult.Success, Grpc.Core.StatusCode.Aborted)]
+    [InlineData(true, ExportResult.Success, Grpc.Core.StatusCode.OutOfRange)]
+    [InlineData(true, ExportResult.Success, Grpc.Core.StatusCode.DataLoss)]
+    [InlineData(true, ExportResult.Failure, Grpc.Core.StatusCode.Internal)]
+    [InlineData(true, ExportResult.Failure, Grpc.Core.StatusCode.InvalidArgument)]
+    [InlineData(true, ExportResult.Failure, Grpc.Core.StatusCode.Unimplemented)]
```

Comment:
Retrying for unimplemented status code seems wrong.

How was this list compiled? The [spec](https://github.com/open-telemetry/opentelemetry-proto/blob/main/docs/specification.md#failures) also mentions this status code to be non-retriable:

|gRPC Code|Retryable?|
|---------|----------|
|CANCELLED|Yes|
|UNKNOWN|No|
|INVALID_ARGUMENT|No|
|DEADLINE_EXCEEDED|Yes|
|NOT_FOUND|No|
|ALREADY_EXISTS|No|
|PERMISSION_DENIED|No|
|UNAUTHENTICATED|No|
|RESOURCE_EXHAUSTED|Only if the server can recover (see below)|
|FAILED_PRECONDITION|No|
|ABORTED|Yes|
|OUT_OF_RANGE|Yes|
|UNIMPLEMENTED|No|
|INTERNAL|No|
|UNAVAILABLE|Yes|
|DATA_LOSS|Yes|

Reply:
> The [spec](https://github.com/open-telemetry/opentelemetry-proto/blob/main/docs/specification.md#failures) also mentions this status code to be non-retriable

Correct and that's what the test confirms. The test is set up in a way that the first response would be one of the status codes from the list and the second one would be `Ok`. 

For e.g. take a look at this case

`InlineData(true, ExportResult.Failure, Grpc.Core.StatusCode.InvalidArgument)`

When the status code is retryable then the expected result should be `ExportResult.Success`. But in this case we assert that the result is `ExportResult.Failure` i.e. the request will not be retried.

This ^ is different than the following case

`[InlineData(true, ExportResult.Success, Grpc.Core.StatusCode.Unavailable)]`

Reply:
Oh okay. I got confused because of the parameter name `shouldRetry`. Could you change it to `useRetryTransmissionHandler`? That makes the intent clear.